### PR TITLE
i#6992 doxygen 1.12: heading and reference workaround

### LINKED ISSUE #6992: Mac doxygen failing with style and reference errors
This just showed up on #6111 and #6990.  Probably Github updated the Doxygen version on the Mac image:

```
CMake Error at /Users/runner/work/dynamorio/dynamorio/api/docs/CMake_rundoxygen.cmake:154 (message):
  *** /usr/local/bin/doxygen failed: ***
  error: Style 'Heading7' redefines \s0.
  error: Style 'Heading8' redefines \s0.
  error: Style 'Heading9' redefines \s0.
  error: Style 'Heading11' redefines \s0.
  error: Style 'Heading12' redefines \s0.
  error: Style 'Heading7' redefines \s0.
  error: Style 'Heading8' redefines \s0.
  error: Style 'Heading9' redefines \s0.
  <...>
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:52:
  warning: unable to resolve reference to 'ref1' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:54:
  warning: unable to resolve reference to 'ref2' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:54:
  warning: unable to resolve reference to 'ref3' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:56:
  warning: unable to resolve reference to 'ref4' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:52:
  warning: unable to resolve reference to 'ref1' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:54:
  warning: unable to resolve reference to 'ref2' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:54:
  warning: unable to resolve reference to 'ref3' for \ref command
  /Users/runner/work/dynamorio/dynamorio/api/docs/tutorial_secdev16.dox:56:
  warning: unable to resolve reference to 'ref4' for \ref command
  error: Style 'Heading11' redefines \s0.
  error: Style 'Heading12' redefines \s0.
  error: Style 'Heading7' redefines \s0.
```

### PR DESCRIPTION
**Pull request by edeiana**
Generating RTF documentation is causing `error: Style 'Heading7' redefines \s0.`
errors with doxygen 1.12. We disable RTF in api/docs/API.doxy.

We fix `warning: unable to resolve reference to 'ref1' for \ref command` issues
using `\anchor ref1` instead of `<a name="ref1"></a>` when citing papers in
"SecDev 2016 Tutorial".

Fixes #6992

### CODE REVIEW

File: api/docs/API.doxy [hunk 1]
```diff
@@ -170,7 +170,7 @@ LATEX_HIDE_INDICES     = NO
 #---------------------------------------------------------------------------
 # configuration options related to the RTF output
 #---------------------------------------------------------------------------
-GENERATE_RTF           = YES
+GENERATE_RTF           = NO
```
Comment (derekbruening) [inline api/docs/API.doxy:173]: Also remove from CMakeLists.txt?
```
  set_directory_properties(PROPERTIES
    ADDITIONAL_MAKE_CLEAN_FILES "html;latex;rtf")
```

Comment (derekbruening) [inline api/docs/API.doxy:173]: And these image versions?
```
api/docs/intro.dox:  \image rtf interpose.png
api/docs/intro.dox:  \image rtf flow-highlevel.png
api/docs/intro.dox:  \image rtf flow.png
api/docs/intro.dox:  \image rtf client.png
api/docs/transp.dox:  \image rtf windows.png
```

**Review by derekbruening — APPROVED** [review]
Thanks for fixing so quickly!

File: api/docs/API.doxy (hunk 1 above)
Comment (edeiana) [reply api/docs/API.doxy:173]: Done.

Comment (edeiana) [reply api/docs/API.doxy:173]: Done.

**Pull request merged**